Tasks/Common/webdeployment-common/azurerestutility.ts

```diff
@@ -213,3 +213,95 @@ export async function getAzureRMWebAppConfigDetails(SPN, webAppName: string, res
     });
     return deferred.promise;
 }
+
+export async function getWebAppAppSettings(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string, obj: Object)
+{
+	if(!deployToSlotFlag) {
+       var requestURL = armUrl + 'subscriptions/' + SPN.subscriptionId + '/resources?$filter=resourceType EQ \'Microsoft.Web/Sites\' AND name EQ \'' + 
+                          webAppName + '\'&api-version=2016-07-01';
+        var accessToken = await getAuthorizationToken(SPN);
+        var headers = {
+            authorization: 'Bearer '+ accessToken
+        };
+        var webAppID = await getAzureRMWebAppID(SPN, webAppName, requestURL, headers);
+        tl.debug('Web App details : ' + webAppID.id);
+        resourceGroupName = webAppID.id.split ('/')[4];
+        tl.debug('AzureRM Resource Group Name : ' + resourceGroupName);
+    }
+
+    var deferred = Q.defer<any>();
+	
+	var accessToken = await getAuthorizationToken(SPN);
+    var headers = {
+        authorization: 'Bearer '+ accessToken
+    };
+	
+	var slotUrl = deployToSlotFlag ? "/slots/" + slotName : "";
+    var configUrl = armUrl + 'subscriptions/' + SPN.subscriptionId + '/resourceGroups/' + resourceGroupName +
+             '/providers/Microsoft.Web/sites/' + webAppName + slotUrl +  '/config/appsettings/list?' + azureApiVersion;
+	
+	tl.debug('Requesting for the Current List of App Settings: ' + configUrl);
+
+	httpObj.send('POST', configUrl, null, headers, (error, response, body) =>{
+		if(error){
+			deferred.reject(error);
+		}
+		else if(response.statusCode === 200) {
```

Comment:
No, the  rest api documentation does not have a nextLink property for the response of the api call.